RiotSwiftUI/Modules/LocationSharing/StartLocationSharing/Coordinator/LocationSharingCoordinator.swift

```diff
@@ -166,17 +166,34 @@ final class LocationSharingCoordinator: Coordinator, Presentable {
     
     // Check if user can send beacon info state event
     private func canShareLiveLocation() -> Bool {
-        guard let myUserId = parameters.roomDataSource.mxSession.myUserId else {
-            return false
-        }
-        
-        let userPowerLevelRawValue = parameters.roomDataSource.roomState.powerLevels.powerLevelOfUser(withUserID: myUserId)
-        
-        guard let userPowerLevel = RoomPowerLevel(rawValue: userPowerLevelRawValue) else {
+        // Tchap: allow live sharing geolocation based on room power levels
+        //
+//        guard let myUserId = parameters.roomDataSource.mxSession.myUserId else {
+//            return false
+//        }
+//        
+//        let userPowerLevelRawValue = parameters.roomDataSource.roomState.powerLevels.powerLevelOfUser(withUserID: myUserId)
+//        
+//        guard let userPowerLevel = RoomPowerLevel(rawValue: userPowerLevelRawValue) else {
+//            return false
+//        }
+//        
+//        return userPowerLevel.rawValue >= RoomPowerLevel.moderator.rawValue
+
+        guard let myUserId = parameters.roomDataSource.mxSession.myUserId,
+              let roomPowerLevels = parameters.roomDataSource.roomState.powerLevels,
+              let userPowerLevel = RoomPowerLevel(rawValue: parameters.roomDataSource.roomState.powerLevels.powerLevelOfUser(withUserID: myUserId)) else {
             return false
         }
         
-        return userPowerLevel.rawValue >= RoomPowerLevel.moderator.rawValue
+        // Tchap: should call `minimumPowerLevelForSendingStateEvent(_ eventType: MXEventType) -> Int` from `MatrixSDK:MXRoomPowerLevels.swift`
+        // but can't because it is inaccessible due to 'internal' protection level
+        //
+        // let liveSharingPowerLevel = parameters.roomDataSource.roomState.powerLevels.minimumPowerLevelForSendingStateEvent(.beaconInfo)
+        //
+        let liveSharingPowerLevel = roomPowerLevels.events[kMXEventTypeStringBeaconInfoMSC3672] as? Int ?? roomPowerLevels.stateDefault
```

Comment:
you should test kMXEventTypeStringBeaconInfo key too.

Reply:
Now, I get live sharing power level from stable value, then unstable value and fallback on default value.